### PR TITLE
fix(Core/Spell): Eye of Gruul healing discount spell

### PR DESCRIPTION
**Pull request by GMKyle**
Fix triggered spell from casting healing spells wearing Eye of Gruul (http://wotlk.cavernoftime.com/item=28823)

##### CHANGES PROPOSED:
-  Improve PR https://github.com/azerothcore/azerothcore-wotlk/pull/1764
-  Benefit all healing classes wearing Eye of Gruul

**TESTS PERFORMED**: Build without errors and tested in-game. (Win10 and Debian 9 servers).

##### HOW TO TEST THE CHANGES:

 - Create healing class lvl >=70
 - .add 28823
 - Cast healing spells until the item procs Healing discount (Could take a while xD)
 - The next healing spell cost should reduced by 450 while Healing discount buff lasts

##### Target branch(es):

Master

### CODE REVIEW

File: data/sql/updates/pending_db_world/rev_1559162254925218831.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1559162254925218831');
+
+DELETE FROM `spell_script_names` WHERE `ScriptName`="spell_pal_item_healing_discount" AND `spell_id`=37705;
```
Comment (FrancescoBorzi) [inline data/sql/updates/pending_db_world/rev_1559162254925218831.sql]: please use single quotes to wrap strings in SQL:

```sql
`ScriptName`='spell_pal_item_healing_discount'
```

File: data/sql/updates/pending_db_world/rev_1559162254925218831.sql [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+INSERT INTO `spell_script_names` (`spell_id`, `ScriptName`) VALUES
+(37705, "spell_item_eye_of_gruul_healing_discount");
```
Comment (FrancescoBorzi) [inline data/sql/updates/pending_db_world/rev_1559162254925218831.sql]: same here

File: src/server/scripts/Spells/spell_item.cpp [hunk 3]
```diff
@@ -1386,6 +1386,73 @@ class spell_item_direbrew_remote : public SpellScriptLoader
...
+                    spell_id = SPELL_PALADIN_ITEM_HEALING_TRANCE;
+                    break;
+                case CLASS_PRIEST:
+                    spell_id = SPELL_PRIEST_ITEM_HEALING_TRANCE;
+                    break;
+                case CLASS_SHAMAN:
+                    spell_id = SPELL_SHAMAN_ITEM_HEALING_TRANCE;
+                    break;
+                default:
+                    return; // ignore for non-healing classes
```
Comment (FrancescoBorzi) [inline src/server/scripts/Spells/spell_item.cpp]: please fix indentation (`case`s should have one more indentation than `switch`)

File: data/sql/updates/pending_db_world/rev_1559162254925218831.sql (hunk 1 above)
Comment (GMKyle) [reply data/sql/updates/pending_db_world/rev_1559162254925218831.sql]: done!!

**Comment by FrancescoBorzi** [comment]
has this been tested?

**Comment by pklloveyou** [comment]
Pass the test and work as expected

**Comment by FrancescoBorzi** [comment]
solved merge conflict, now ready to be merged.

good job @GMKyle and @pklloveyou

**Comment by Stoabrogga** [comment]
So, if merging this can we close PR #1764?

**Comment by GMKyle** [comment]
Yes, it can be closed

**Pull request merged**